catalyst/runners/self_supervised.py

```diff
@@ -135,6 +136,7 @@ def __init__(
         augemention_prefix: str = "augment",
         projection_prefix: str = "projection",
         embedding_prefix: str = "embedding",
+        encoders: Optional[List[str]] = None,
```

Comment:
do we really need it?
why could not we just work with the KeyValue model, as you already write by default? iterating over the keys?

Reply:
I want to fix it here in that case; we will have a convention that if our model is Map[str,nn.Module] each module is an encoder model.

And could you suggest a smart way how to check that model is a Map?

Reply:
What do you think about [the implemention](https://github.com/Nimrais/catalyst/blob/byol-implemention/catalyst/runners/self_supervised.py#L175-L178)?